vars/neoDeploy.groovy

```diff
@@ -8,222 +11,218 @@ import com.sap.piper.tools.ToolDescriptor
+            .mixin(parameters, PARAMETER_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .withMandatoryProperty('neo', 'Error in neoDeploy: neo not configured.')
+            .withMandatoryProperty('source', 'Error in neoDeploy: source not configured.')
+            .withMandatoryProperty('deployMode', 'Error in neoDeploy: deployMode not configured.')
+            .withMandatoryProperty('warAction', 'Error in neoDeploy: warAction not configured.')
             .use()
 
         utils.pushToSWA([
-            step: STEP_NAME,
-            stepParam1: configuration.deployMode == 'mta'?'mta':'war', // ['mta', 'warParams', 'warPropertiesFile']
-            stepParam2: configuration.warAction == 'rolling-update'?'blue-green':'standard', // ['deploy', 'deploy-mta', 'rolling-update']
+            step      : STEP_NAME,
+            stepParam1: configuration.deployMode == 'mta' ? 'mta' : 'war', // ['mta', 'warParams', 'warPropertiesFile']
+            stepParam2: configuration.warAction == 'rolling-update' ? 'blue-green' : 'standard', // ['deploy', 'deploy-mta', 'rolling-update']
             stepParam3: parameters?.script == null,
-            stepParam4: ! stepCompatibilityConfiguration.isEmpty(),
+            stepParam4: !stepCompatibilityConfiguration.isEmpty(),
         ], configuration)
 
-        def archivePath = configuration.archivePath
-        if(archivePath?.trim()) {
-            if (!fileExists(archivePath)) {
-                error "Archive cannot be found with parameter archivePath: '${archivePath}'."
+        ToolDescriptor neo = new ToolDescriptor('SAP Cloud Platform Console Client', 'NEO_HOME', 'neoHome', '/tools/', 'neo.sh', null, 'version')
+        ToolDescriptor java = new ToolDescriptor('Java', 'JAVA_HOME', '', '/bin/', 'java', '1.8.0', '-version 2>&1')
+
+        if (configuration.neo.credentialsId) {
+            withCredentials([usernamePassword(
+                credentialsId: configuration.neo.credentialsId,
+                passwordVariable: 'NEO_PASSWORD',
+                usernameVariable: 'NEO_USERNAME')]) {
+
+                assertPasswordRules(NEO_PASSWORD)
+                String neoExecutable = neo.getToolExecutable(script, configuration)
+                NeoCommandHelper neoCommandHelper = new NeoCommandHelper(
+                    script,
+                    configuration.deployMode,
+                    configuration.neo,
+                    neoExecutable,
+                    NEO_USERNAME,
+                    NEO_PASSWORD,
+                    configuration.source
+                )
+
+                dockerExecute(
+                    script: script,
+                    dockerImage: configuration.get('dockerImage'),
+                    dockerEnvVars: configuration.get('dockerEnvVars'),
+                    dockerOptions: configuration.get('dockerOptions')
+                ) {
+
+                    neo.verify(this, configuration)
```

Comment:
I would like to suggest to rely on a properly configured system e.g. provided via docker. When running on docker images tested by us is there still the need for having those checks? How important are use cases running with the tools installed on a local Jenkins. IMO we should guide users to using docker images. IMO the vision is to run the lib on a dockerized Jenkins delivered by us with additional tool specific docker images we also tested.

Reply:
Totally agree, but let's do it in a follow-up